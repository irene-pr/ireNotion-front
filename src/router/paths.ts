const paths = {
  home: "/",
  login: "/login",
  register: "/register",
  userBoard: "/userBoard",
};

export default paths;
