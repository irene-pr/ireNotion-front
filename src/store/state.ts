const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
