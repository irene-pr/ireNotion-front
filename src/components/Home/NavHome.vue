<template>
  <nav class="nav__user-area">
    <router-link :to="paths.login">
      <button class="nav__button nav__button--login" disabled>
        LOG <span>IN</span>
      </button>
    </router-link>
    <router-link :to="paths.login">
      <button class="nav__button nav__button--signup">
        SIGN <span>UP</span>
      </button>
    </router-link>
  </nav>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import paths from "../../router/paths";

export default defineComponent({
  name: "UserArea",
  data() {
    return {
      paths,
    };
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.nav__user-area {
  display: flex;
  position: fixed;
  z-index: 2;
}
.nav__button {
  position: absolute;
  top: 14px;
  @extend %button;

  &--login {
    @extend %button--login;
    left: 14px;
  }
  &--signup {
    @extend %button--signup;
    left: 116px;
  }
  span {
    @extend %button--span;
    color: $theme-pink;
  }
}
</style>
