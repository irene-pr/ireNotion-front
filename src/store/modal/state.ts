import { IModalState } from "@/types/store";

const state = (): IModalState => ({
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
});

export default state;
