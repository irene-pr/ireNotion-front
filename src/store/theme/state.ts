import { IThemeState } from "@/types/store";

const state = (): IThemeState => ({
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
});

export default state;
