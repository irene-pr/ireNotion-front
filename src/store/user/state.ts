import { IUserState } from "@/types/store";

const state = (): IUserState => ({
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  isLoading: false,
});

export default state;
