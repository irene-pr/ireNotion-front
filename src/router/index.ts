import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import paths from "@/router/paths";
import Home from "../views/Home.vue";
import Login from "../views/Login.vue";
import Register from "../views/Register.vue";
import BoardPage from "../views/BoardPage.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: paths.home,
    name: "Home",
    component: Home,
  },
  {
    path: paths.login,
    name: "Login",
    component: Login,
  },
  {
    path: paths.register,
    name: "Register",
    component: Register,
  },
  {
    path: paths.userBoard,
    name: "BoardPage",
    component: BoardPage,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
