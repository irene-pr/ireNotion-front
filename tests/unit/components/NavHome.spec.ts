import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import NavHome from "@/components/Home/NavHome.vue";
import state from "@/store/state";
import router from "@/router";

describe("Given a NavHome component", () => {
  const redirectToLoginMock = jest.fn();
  const redirectToRegisterMock = jest.fn();
  const store = createStore({
    state() {
      return state;
    },
    getters: {
      redirectToLogin: redirectToLoginMock,
      redirectToRegister: redirectToRegisterMock,
    },
  });

  describe("When it is rendered", () => {
    test("Then it renders", () => {
      mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
    });
    test("Then the it matches the snapshot", () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Then it renders two buttons", () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const buttons = wrapper.findAll("button");

      expect(buttons).toHaveLength(2);
    });
    test("Then it renders the login button", () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const button = wrapper.find("button.nav__button--login");

      expect(button.text()).toContain("LOG IN");
    });
    test("Then it renders the signup button", () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const button = wrapper.find("button.nav__button--signup");

      expect(button.text()).toContain("SIGN UP");
    });
  });
  describe("When you click on the login button", () => {
    test("Then the redirectToLogin getter is called", async () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const button = wrapper.find("button.nav__button--login");
      button.trigger("click");

      expect(redirectToLoginMock).toHaveBeenCalled();
    });
  });
  describe("When you click on the signup button", () => {
    test("Then the redirectToRegister getter is called", async () => {
      const wrapper = mount(NavHome, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const button = wrapper.find("button.nav__button--signup");
      button.trigger("click");

      expect(redirectToRegisterMock).toHaveBeenCalled();
    });
  });
});
