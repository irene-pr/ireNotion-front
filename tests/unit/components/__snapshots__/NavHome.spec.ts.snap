// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NavHome component When it is rendered Then the it matches the snapshot 1`] = `
<nav
  class="nav__user-area"
>
  <button
    class="nav__button nav__button--login"
  >
     LOG 
    <span>
      IN
    </span>
  </button>
  <button
    class="nav__button nav__button--signup"
  >
     SIGN 
    <span>
      UP
    </span>
  </button>
</nav>
`;
